### PR TITLE
Better document CALL ... YIELD ... (and WHERE ...)

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 1]
```diff
@@ -6,7 +6,28 @@
 The `CALL` clause is used to call a procedure deployed in the database.
 --
 
-The examples showing how to use arguments when invoking procedures all use the following procedure:
+Procedures are called using the `CALL` clause.
+
+Each procedure call needs to specify all required procedure arguments.
+This may be done either explicitly, by listing them in parenthesis and separated by commas after the procedure name, or implicitly by using available query parameters as procedure call arguments.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: I suggest changing 

>  by listing them in parenthesis and separated by commas after the procedure name

to

> using a comma-separated list wrapped in parentheses after the procedure name

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 2]
```diff
@@ -6,7 +6,28 @@
...
+The latter form is only available when the whole query consists of a single `CALL` clause (i.e. in a standalone procedure call).
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: I suggest using this paragraph to clearly define `standalone procedure call`, perhaps:

> The latter form is available only in a so-called standalone call, when the whole query consists of a single `CALL` clause.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 3]
```diff
@@ -6,7 +6,28 @@
...
+
+Most procedures return a stream of records with a fixed set of result fields, similar to how running a Cypher query returns a stream of records, too.
+The `YIELD` sub clause is used to explicitly select which of these result fields are returned as newly bound variables from the procedure call to the user or for further processing by the remaining query.
+Each yielded result field may optionally be renamed using aliasing (i.e. `resultFieldName AS newName`).
+All new variables bound by a procedure call are added to the set of variables already bound in the current scope.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc:18]: All -> Any

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 4]
```diff
@@ -6,7 +6,28 @@
...
+It is an error if a procedure call tries to rebind an already bound variable (i.e. a procedure call cannot shadow a variable that is already bound in the current scope).
+
+Inside a larger query, the records returned from a procedure call with explicit `YIELD` may be further filtered using a `WHERE` sub clause followed by a predicate (similar to `WITH ... WHERE ...`).
+
+In a standalone procedure call only, `YIELD` may be omitted.
+In this case, all result fields are yielded as newly bound variables from the procedure call to the user or for further processing by the remaining query.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: This is in the context of a standalone procedure call, so there is no remaining query to speak of.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 5]
```diff
@@ -6,7 +6,28 @@
...
+
+Calling a `VOID` procedures may only have a side effect and will not produce any result records and thus does neither allow nor require the use of `YIELD`.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: `VOID` procedure has not yet been defined here. See above comment about defining them.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 6]
```diff
@@ -6,7 +6,28 @@
...
 --
 
-The examples showing how to use arguments when invoking procedures all use the following procedure:
+Procedures are called using the `CALL` clause.
+
+Each procedure call needs to specify all required procedure arguments.
+This may be done either explicitly, by listing them in parenthesis and separated by commas after the procedure name, or implicitly by using available query parameters as procedure call arguments.
+The latter form is only available when the whole query consists of a single `CALL` clause (i.e. in a standalone procedure call).
+
+Most procedures return a stream of records with a fixed set of result fields, similar to how running a Cypher query returns a stream of records, too.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc:15]: I suggest defining `VOID` procedures here, perhaps:

> Procedures that do not declare any result fields are called `VOID` procedures.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc (hunk 5 above)
Comment (Mats-SX) [reply manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: I suggest changing this to 

> Calling a `VOID` procedure may have side effects, but will not produce any result records and thus does neither allow nor require the use of `YIELD`.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 7]
```diff
@@ -6,7 +6,28 @@
...
+
+Inside a larger query, the records returned from a procedure call with explicit `YIELD` may be further filtered using a `WHERE` sub clause followed by a predicate (similar to `WITH ... WHERE ...`).
+
+In a standalone procedure call only, `YIELD` may be omitted.
+In this case, all result fields are yielded as newly bound variables from the procedure call to the user or for further processing by the remaining query.
+
+Calling a `VOID` procedures may only have a side effect and will not produce any result records and thus does neither allow nor require the use of `YIELD`.
+Calling a `VOID` procedure in the middle of a larger query will simply pass on each input record (i.e. it acts like `WITH *`).
+
+The following examples show how to pass arguments to and how to yield result fields from a procedure call.
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: May omit the second `how to`.

File: manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallTest.scala [hunk 8]
```diff
@@ -70,6 +70,17 @@ class CallTest extends DocumentingTestBase with QueryStatisticsTestSupport with
       assertions = (p) => assert(p.nonEmpty) )
   }
 
+  @Test def call_a_procedure_and_filter_its_results() {
+    testQuery(
+      title = "Call a procedure and filter its results",
+      text = "This calls the built-in procedure 'db.labels' to count all in-use labels in the database that contain the word 'User'",
```
Comment (Mats-SX) [inline manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CallTest.scala]: I suggest replacing apostrophes (') with backticks (`).

**Review by Mats-SX — CHANGES_REQUESTED** [review]
Nice docs. I'd like to see clearer definitions of standalone calls and void procedures.

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc (hunk 6 above)
Comment (petraselmer) [reply manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc:15]: I would leave out the ", too" text (doesn't add anything)

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 9]
```diff
@@ -6,7 +6,28 @@
...
 
-The examples showing how to use arguments when invoking procedures all use the following procedure:
+Procedures are called using the `CALL` clause.
+
+Each procedure call needs to specify all required procedure arguments.
+This may be done either explicitly, by listing them in parenthesis and separated by commas after the procedure name, or implicitly by using available query parameters as procedure call arguments.
+The latter form is only available when the whole query consists of a single `CALL` clause (i.e. in a standalone procedure call).
+
+Most procedures return a stream of records with a fixed set of result fields, similar to how running a Cypher query returns a stream of records, too.
+The `YIELD` sub clause is used to explicitly select which of these result fields are returned as newly bound variables from the procedure call to the user or for further processing by the remaining query.
```
Comment (petraselmer) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: "sub clause" -> "sub-clause" throughout

Comment (petraselmer) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: "newly bound" -> "newly-bound" (all instances)

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 10]
```diff
@@ -6,7 +6,28 @@
...
+Each yielded result field may optionally be renamed using aliasing (i.e. `resultFieldName AS newName`).
+All new variables bound by a procedure call are added to the set of variables already bound in the current scope.
+It is an error if a procedure call tries to rebind an already bound variable (i.e. a procedure call cannot shadow a variable that is already bound in the current scope).
```
Comment (petraselmer) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: "an already bound" -> "a previously bound" (all instances)

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 11]
```diff
@@ -6,7 +6,28 @@
...
+
+Inside a larger query, the records returned from a procedure call with explicit `YIELD` may be further filtered using a `WHERE` sub clause followed by a predicate (similar to `WITH ... WHERE ...`).
```
Comment (petraselmer) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: "with explicit `YIELD`" -> "with an explicit `YIELD`"

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc [hunk 12]
```diff
@@ -6,7 +6,28 @@
...
+
+In a standalone procedure call only, `YIELD` may be omitted.
```
Comment (petraselmer) [inline manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: "In a standalone procedure call only, `YIELD` may be omitted."
->
"`YIELD` may be omitted in a standalone procedure call."

File: manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc (hunk 5 above)
Comment (petraselmer) [reply manual/cypher/cypher-docs/src/docs/dev/ql/call/index.asciidoc]: I concur with @Mats-SX 's comment, but with the further additional change:
"thus does neither allow nor require the use of `YIELD`"
->
"thus neither allows nor requires the use of YIELD"

**Comment by petraselmer** [comment]
@boggle please let me know once you have addressed the various comments etc. I will then undertake to get this content into the relevant repo, and close this PR.

**Comment by boggle** [comment]
@petraselmer I updated the PR; could please you take this on and get the test and the manual content to the right place?

**Comment by petraselmer** [comment]
Closing, as superseded by https://github.com/neo4j/neo4j-documentation/pull/1

**Pull request closed without merging**